apollo-api/src/main/java/com/apollographql/apollo/api/ResponseField.java

```diff
@@ -277,30 +277,47 @@ private String orderIndependentKey(Map<String, Object> objectMap, Operation.Vari
       }
     });
     StringBuilder independentKey = new StringBuilder();
+    independentKey.append("{");
     for (int i = 0; i < sortedArguments.size(); i++) {
       Map.Entry<String, Object> argument = sortedArguments.get(i);
       if (argument.getValue() instanceof Map) {
         //noinspection unchecked
         final Map<String, Object> objectArg = (Map<String, Object>) argument.getValue();
-        boolean isArgumentVariable = isArgumentValueVariableType(objectArg);
         independentKey
+            .append("\"")
             .append(argument.getKey())
-            .append(":")
-            .append(isArgumentVariable ? "" : "[")
-            .append(orderIndependentKey(objectArg, variables))
-            .append(isArgumentVariable ? "" : "]");
+            .append("\":")
+            .append(orderIndependentKey(objectArg, variables));
       } else {
-        independentKey.append(argument.getKey())
-            .append(":")
-            .append(argument.getValue().toString());
+        independentKey
+            .append("\"")
+            .append(argument.getKey())
+            .append("\":")
+            .append(asJsonValue(argument.getValue()));
       }
       if (i < sortedArguments.size() - 1) {
         independentKey.append(",");
       }
     }
+    independentKey.append("}");
     return independentKey.toString();
   }
 
+  private String asJsonValue(Object o) {
+    if (o instanceof Boolean
+        || o instanceof Byte
+        || o instanceof Character
+        || o instanceof Double
+        || o instanceof Float
+        || o instanceof Integer
+        || o instanceof Long
+        || o instanceof Short) {
+      return o.toString();
```

Comment:
1. It can be simplified, just check for `instanceof Number` and it will cover `Byte, Double, Float, Integer, Long, Short`.
2. Not sure that `Character` is supported json value type, should it be wrapped in ""?

Reply:
Good point, fixed in ec8b540